orca-queue/src/main/kotlin/com/netflix/spinnaker/orca/q/handler/ContinueParentStageHandler.kt

```diff
@@ -46,14 +43,23 @@ class ContinueParentStageHandler(
 
   override fun handle(message: ContinueParentStage) {
     message.withStage { stage ->
-      if (stage.allBeforeStagesComplete()) {
-        when {
-          stage.hasTasks() -> stage.runFirstTask()
-          else -> queue.push(CompleteStage(stage))
+      if (message.phase == STAGE_BEFORE) {
+        if (stage.allBeforeStagesComplete()) {
+          when {
+            stage.hasTasks() -> stage.runFirstTask()
+            else             -> queue.push(CompleteStage(stage))
+          }
+        } else if (!stage.anyBeforeStagesFailed()) {
```

Comment:
What if this is true? Do we have to handle the failure here?

Reply:
I think the `ContinueParentStage` message is only sent in the case where a synthetic stage doesn't fail (the failing synthetic stage would end up completing the execution).

I think this handles the case where a `ContinueParentStage` message is sent -> a sibling synthetic stage fails -> this handler picks up the `ContinueParentStage` message and should do nothing because the execution is going to be completed by a different handler.